lib/web_ui/dev/felt

```diff
@@ -56,9 +56,45 @@ install_deps() {
 KERNEL_NAME=`uname`
 if [[ $KERNEL_NAME == *"Darwin"* ]]
 then
-  echo "Running on MacOS. Increase the user limits"
-  ulimit -n 50000
-  ulimit -u 4096
+  echo "Running on MacOS. Will check the file and user limits."
+  # Disable exit if the commands fails. We want to give more actionable
+  # error message.
+  set +e
+  ULIMIT_FILES=`ulimit -n`
+  # Increase the file limit if it is low. Note that these limits are changed
+  # only for this script (for this shell). After felt execution is completed
+  # no change is required to reset the original shell.
+  if [[ $ULIMIT_FILES < 50000 ]]
+  then
+    echo "File limits too low increasing the file limits"
+    # Get the max file limit.
+    MAX_ULIMIT_FILES=`launchctl limit maxfiles | sed -e 's/^[[:space:]]*//' | sed 's/ \{1,\}/ /g' | cut -d' ' -f2`
+    if [[ $MAX_ULIMIT_FILES < 50000 ]]
+    then
+      # Increase the maximum file limit.
+      sudo launchctl limit maxfiles 50000 200000
+    fi
+    ERROR=$(ulimit -n 50000 2>&1 >/dev/null)
+    if [[ ! -z $ERROR ]]
+    then
+      echo "Problem changing the file limit. Please try to reboot. error: \n$ERROR"
+      exit 1
```

Comment:
I would actually not exit, but print the warning and continue running the test. Chances are if you are running one test locally, you may not need such a high limit.

Reply:
done.